vocab/security/vocabulary.yml

```diff
@@ -235,7 +235,7 @@ property:
   - id: cryptosuite
     label: Cryptographic suite
     domain: sec:DataIntegrityProof
-    range: xsd:string
+    range: sec:cryptosuiteString
```

Comment:
That is not correct in the vocabulary: the value of `range` must be a class or a new datatype. Do we really want to go down the (tedious!) process of formally define a new datatype for `cryptosuiteString`?

I would propose to leave the range a string, and add a comment (or part of the `cryptosuite` property definition in the spec) what the constraints on the value are. But do not treat it as a datatype with a separate URL.

Reply:
Just to add to my previous remark. If the WG _does_ decide to define a new datatype, what I would think we should do is as follows.

- I do not think we should bother extending the yml2vocab script to add a datatype definition feature in general; this is a very rare occurrence, and this tool is not aimed at being a universal RDF tool. Also, the datatype definition is _not_ part of the RDF Vocabulary Graph, in other words, it is not something we would add to the Turtle and JSON-LD files, only the HTML version. As a consequence, what I would propose to do is to add the formal definition to the integrity spec (see below), and add a separate section into the Template html file that refers back to the definition in the integrity spec, a bit like all the others. I am happy to do that, once the references in the integrity spec is available.
- The formal definition of a datatype requires the formal definition of the lexical space, value space, etc. Even for a type that is relatively simple it is, as I said, tedious. I think the formal definition of the [rdf:JSON Datatype](https://www.w3.org/TR/json-ld11/#the-rdf-json-datatype) is a good example for what should be added to the integrity spec (probably in a normative appendix, to avoid breaking the flow of the main text).

Reply:
>  Do we really want to go down the (tedious!) process of formally define a new datatype for cryptosuiteString?

Yes, we do, because if we don't do that, the resolution of using long cryptosuite names falls apart due to the burden it places on low-bandwidth environments. In other words, the pain felt in defining a new type will be on the Editors. If we don't define this type, the pain felt will be on every implementer out there in a low bandwidth environment. I'll volunteer to define the data type, thanks for the pointer to rdf:JSON... we'll model it after that.

Reply:
> Yes, we do, because if we don't do that, the resolution of using long cryptosuite names falls apart due to the burden it places on low-bandwidth environments. In other words, the pain felt in defining a new type will be on the Editors. If we don't define this type, the pain felt will be on every implementer out there in a low bandwidth environment. I'll volunteer to define the data type, thanks for the pointer to rdf:JSON... we'll model it after that.

Fine. Then let us do it, but we should follow something like https://github.com/w3c/vc-data-integrity/pull/162#discussion_r1285419464 to make it proper. It is tedious but not too much after all; the only important point is to define more formally the subset of strings that can be used as a datatype value. The rest is obvious (and the rdf:JSON stuff could indeed be used as a pattern).

I am o.k. to expand the formal vocabulary with the datatype stuff once the core spec has it, either in this PR or (probably better) in another one.

Reply:
@iherman I have added the definition of cryptosuiteString, based on the link to provided for rdf:JSON, in this commit: https://github.com/w3c/vc-data-integrity/pull/162/commits/40be121985806f03bc07dbef7a253e91b587d737

Requesting a re-review from you now.

Reply:
Requesting a re-review from @TallTed as well because I added a bunch of text that's hard to read and I need his help to make it suck less. :)

Reply:
Re-reviewed. Not too horrible. Editorial changes requested.